### PR TITLE
Merge CHERI JNI support bits

### PR DESCRIPTION
**Pull request by brooksdavis**

### CODE REVIEW

File: sys/mips/cheri/ccall.S [hunk 1]
```diff
@@ -579,6 +604,12 @@ CCall_wrong_regs:
 
 #endif /* CPU_CHERI_HW_CCALL_CHECKS */
 
+CCall_wrong_cc:
+	dli	k0, ((CHERI_EXCCODE_SW_CCALLREGS << 8) | 1)
```
Comment (davidchisnall) [inline sys/mips/cheri/ccall.S]: We might want to allocate a new code here.

File: lib/libc/mips/string/memcpy_c.S [hunk 2]
```diff
@@ -75,6 +75,15 @@ memcpy_c:
 	andi     $12, $v0, CAP_ALIGN_MASK # t4 = dst % CAP_SIZE
           CToPtr   $v0, $c0, $c3       # Return ptr in integer return
 	andi     $13, $v1, CAP_ALIGN_MASK # t5 = src % CAP_SIZE
+
+	CGetPerm $a5, $c3
```
Comment (davidchisnall) [inline lib/libc/mips/string/memcpy_c.S]: This is quite ugly and was intended as a quick hack.  It's probably okay for now, but I don't like it.

**Comment by brooksdavis** [comment]
@rwatson any opinions on these changes?  we'd like to merge them before I start the merge from upstream FreeBSD

**Comment by davidchisnall** [comment]
Thanks!

**Pull request merged**